src/Aspire.Hosting/Extensions/ResourceBuilderExtensions.cs

```diff
@@ -332,6 +332,30 @@ private static void ApplyEndpoints<T>(this IResourceBuilder<T> builder, IResourc
         return builder.WithAnnotation(annotation);
     }
 
+    /// <summary>
+    /// Changes an existing endpoint's host port setting.
+    /// </summary>
+    /// <param name="builder">Resource builder for resource with endpoints.</param>
+    /// <param name="endpointName">Name of endpoint to change.</param>
+    /// <param name="hostPort">Host port</param>
+    /// <returns></returns>
+    public static IResourceBuilder<T> WithEndpointHostPost<T>(this IResourceBuilder<T> builder, string endpointName, int hostPort) where T: IResourceWithEndpoints
```

Comment:
nit: Formatting.

Not nit: Are we going to have a ton of these methods? What happened to the callback API?

Reply:
I put this up for discussion. I'm concerned that if we name this WithEndpoint it is confusing because the other WithEndpoint methods create an endpoint whereas this would throw if it doesn't existing (although I guess we can throw up a malformed endpoint and the validate it after the callback finishes.

I think the UseBlobPort and friends is useful because the dev doesn't need to know what the names of the endpoints are in order to change the port.

Thoughts?

Reply:
Then give the callback method a crazy name 😄

Reply:
I'll stick with `WithEndpoint(...)`. I made it so it creates the endpoint if it doesn't exist. I also added `WithImageX(...)` extension methods.